### ISSUE TITLE
`redundantSelf` incorrectly adds `self.` to keypath not referencing its own variable.

### ISSUE THREAD

**Issue by lhunath** [nicklockwood/SwiftFormat#1911]
```swift
class UserScreenPresenter: ScreenPresenter {
    func onAppear() {
        self.sessionInteractor.stage.compactMap(\.?.session).latestValues(on: .main)
    }

    private var session: Session?
}
```

```bash
$ swiftformat --self insert --selfrequired -v test.swift
Running SwiftFormat...
Reading swift-version file at /Users/hubstaff/Developer/hubstaff/ios/.swift-version (version 5.8)
Formatting /Users/hubstaff/Developer/hubstaff/ios/test.swift
-- rules applied: redundantSelf
Writing /Users/hubstaff/Developer/hubstaff/ios/test.swift
SwiftFormat completed in 0.01s.
1/1 files formatted.
```

It appears that SwiftFormat incorrectly assumes the keypath is referring to a property on `UserScreenPresenter` while this is not the case.

**Comment by nicklockwood** [comment]
@lhunath is `.?.` a typo? I've never seen that construct before?

**Comment by lhunath** [comment]
Not a typo, :-). Optional dereference in keypath

**Comment by nicklockwood** [comment]
TIL!

**Comment by nicklockwood** [comment]
@lhunath fixed in 0.55.2
